src/__tests__/lib/command/util/schema-util.test.ts

```diff
@@ -358,3 +395,106 @@ test('chooseSchemaAppFn uses correct endpoint to list schema apps', async () =>
 
 	expect(apiSchemaListMock).toHaveBeenCalledExactlyOnceWith()
 })
+
+describe('getSchemaAppEnsuringOrganization', () => {
+	const defaultFlags = { json: false, yaml: false }
+
+	it('uses get when the app already has an organization', async () => {
+		apiSchemaGetMock.mockResolvedValueOnce(schemaAppWithOrganization)
+
+		const { schemaApp, organizationWasUpdated } =
+			await getSchemaAppEnsuringOrganization(commandMock, 'has-organization', defaultFlags)
+
+		expect(schemaApp).toBe(schemaAppWithOrganization)
+		expect(organizationWasUpdated).toBe(false)
+
+		expect(apiSchemaListMock).toHaveBeenCalledExactlyOnceWith()
+		expect(apiSchemaGetMock).toHaveBeenCalledExactlyOnceWith('has-organization')
+
+		expect(fatalErrorMock).not.toHaveBeenCalled()
+		expect(chooseOrganizationMock).not.toHaveBeenCalled()
+		expect(apiOrganizationsGetMock).not.toHaveBeenCalled()
+		expect(cloneMock).not.toHaveBeenCalled()
+	})
+
+	it('uses get when no app found in list', async () => {
+		apiSchemaGetMock.mockResolvedValueOnce(schemaAppNotInList)
+
+		const { schemaApp, organizationWasUpdated } =
+			await getSchemaAppEnsuringOrganization(commandMock, 'not-in-list', defaultFlags)
+
+		expect(schemaApp).toBe(schemaAppNotInList)
+		expect(organizationWasUpdated).toBe(false)
+
+		expect(apiSchemaListMock).toHaveBeenCalledExactlyOnceWith()
+		expect(apiSchemaGetMock).toHaveBeenCalledExactlyOnceWith('not-in-list')
+
+		expect(fatalErrorMock).not.toHaveBeenCalled()
+		expect(chooseOrganizationMock).not.toHaveBeenCalled()
+		expect(apiOrganizationsGetMock).not.toHaveBeenCalled()
+		expect(cloneMock).not.toHaveBeenCalled()
+	})
+
+	it('sets and organization when app found without an organization', async () => {
+		chooseOrganizationMock.mockResolvedValueOnce('organization-id-1')
+		apiOrganizationsGetMock.mockResolvedValueOnce(organization1)
+		// TODO: maybe use an "updated organization" here and in corresponding expect below
```

Comment:
Just checking: this `TODO` is really for future work and not just an idea that you considered and rejected, right?

Reply:
Didn't mean to leave that in. Removed now.